lib/resty/memcached.lua

```diff
@@ -736,4 +771,51 @@ function _M.close(self)
 end
 
 
+function _M.init_pipeline(self, n)
+    if self._reqs then 
+        return "already init pipeline"
+    end
+
+    if n and type(n) ~= 'number' then
+        return "buffer size is number type"
+    end
+    self._reqs = {
+        new_tab(n or 4, 0),
+        new_tab(n or 4, 0),
+    }
+    return nil
+end
+
+
+function _M.cancel_pipeline(self)
+    self._reqs = nil
+end
+
+
+function _M.commit_pipeline(self)
+    local reqs = rawget(self, "_reqs")
+    if not reqs then
+        return nil, "no pipeline"
+    end
+    local sock = self.sock
+    if not sock then
+        return nil, "not initialized"
+    end
+
+    if #self._reqs[1] == 0 then
+        return nil, "no more cmds"
+    end
+    local bytes, err = sock:send(concat(reqs[1]))
```

Comment:
```suggestion
    local bytes, err = sock:send(reqs[1])
```

The `concat` can be omitted for better performance.